.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,137 @@
+name: Go Test Workflow
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+env:
+  TYK_IB_STORAGE_STORAGETYPE: file
+  
+permissions:
+  contents: read
+  
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Tyk Identity Broker
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+      - name: Fetch base branch
+        if: ${{ github.event_name == 'pull_request' }}
+        run: git fetch origin ${{ github.base_ref }}
+      - name: golangci-lint
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          args: --out-format=checkstyle:golanglint.xml --timeout=600s --max-issues-per-linter=0 --max-same-issues=0 --new-from-rev=origin/${{ github.base_ref }}
+      - uses: actions/upload-artifact@v2
+        with:
+          name: golangcilint
+          retention-days: 1
+          path: |
+            golanglint.xml
+  
+  ci-test:
+    name: '${{ matrix.databases }}'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        databases:
+          - mongo-mgo
+          - mongo-official
+          - file
+        redis-version: [5]
+        mongodb-version: [4.2]
+    steps:
+     - name: Setup Go
+       uses: actions/setup-go@v2
+       with:
+          go-version: '^1.19'
+    
+     - name: Checkout Code
+       uses: actions/checkout@v4
+       with:
+        ref: ${{ github.event.pull_request.head.ref }}
```

Comment:
This will be empty on push and for PRs, this defaults to the reference or SHA for that event anyway.